pkg/standalone-utils/test/ProtocolIdRegistry.test.ts

```diff
@@ -0,0 +1,98 @@
+        ({ instance: vault, authorizer } = await Vault.create({ admin }));
+        registry = await deploy('ProtocolIdRegistry', {
+            args: [vault.address]
+        })
+    });
+
+    describe('Constructor', () => {
+        it('events are emitted for protocols initialized in the constructor', async () => {
+            expect(
+                await deploy('ProtocolIdRegistry', {
+                    args: [vault.address]
+                })).to.emit('ProtocolIdRegistry', 'ProtocolIdRegistered');
+        });
+
+        context('Aave v1 protocol is registered with protocol id 0', async () => {
+            it('Protocol Id is valid', async () => {
+                expect(await registry.isValidProtocolId(0)).to.equal(true);
+            });
+
+            it('Protocol name is correct', async () => {
+                expect(await registry.getProtocolName(0)).to.equal("Aave v1");
+            });
+        })
+    });
+
+    describe('Registration', () => {
+        let newProtocolId: number = 1000000000; 
+        let newProtocolName: string = "Test Protocol";
+        let transactionReceipt: any;
+        sharedBeforeEach('grant permissions', async () => {
+            await authorizer
+                .connect(admin)
+                .grantPermissions([actionId(registry, 'registerProtocolId')],
+                authorizedUser.address, [
+                    registry.address,
+                ]);
+        });
+
+        context('authorized user', async () => {
+            sharedBeforeEach('register protocol', async () => {
+                transactionReceipt = await ( 
+                    await registry.connect(authorizedUser).registerProtocolId(newProtocolId, newProtocolName)
+                ).wait();
+            })
+
+            it('event emitted', async () => {
+                expectEvent.inReceipt(transactionReceipt, 'ProtocolIdRegistered', {
+                    protocolId: newProtocolId,
+                    name: newProtocolName,
+                  });
+            });
+
+            it('new ID is valid', async () => {
+               expect(await registry.isValidProtocolId(newProtocolId)).to.equal(true);
+            });
+            it('name matches ID', async () => {
+                expect(await registry.getProtocolName(newProtocolId)).to.equal(newProtocolName);
+            });
+            it('reverts when registering existing ID', async () => {
+                await expect(registry.connect(authorizedUser).registerProtocolId(0, "Test Protocol")).to.be.revertedWith('Protocol ID already registered');
+            });
+            it('searching for name in non-existent protocol ID', async () => {
+                await expect(registry.getProtocolName(MAX_UINT256)).to.be.revertedWith('Non-existent protocol ID');
+            });
+            it('check non-valid ID',async () => {
+                expect( await registry.isValidProtocolId(MAX_UINT256)).to.equal(false);
+            });
```

Comment:
These two tests should go outside this `describe` block, since they are not testing registration but rather queries of unregistered ids.